crypto/pkcs11key/key.go

```diff
@@ -222,10 +283,9 @@ func (ps *Key) getPrivateKey(module ctx, session pkcs11.SessionHandle, label str
 	return privateKeyHandle, nil
 }
 
-// Get the public key matching a private key
-// TODO: Add support for non-RSA keys, switching on CKA_KEY_TYPE
-func getPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkcs11.ObjectHandle) (rsa.PublicKey, error) {
-	var noKey rsa.PublicKey
+// Get the public key matching an RSA private key
+func (ps *Key) getRSAPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkcs11.ObjectHandle) (interface{}, error) {
```

Comment:
You made this function a method on `Key`, but it doesn't actually access any members of `Key`. My understanding of Go idiom is that functions are preferred in such cases.